qa/rpc-tests/test_framework/util.py

```diff
@@ -129,17 +129,19 @@ def sync_blocks(rpc_connections, *, wait=1, timeout=60):
     one node already synced to the latest, stable tip, otherwise there's a
     chance it might return before all nodes are stably synced.
```

Comment:
This change might make the problem less likely to happen, but it still could happen. If none of the nodes in the set are fully synced when the `sync_blocks` call is made, then `maxheight` will be set to some height less than the final stable height, and the call return could before the nodes are all synced.